csharp/src/Google.Protobuf/WellKnownTypes/AnyPartial.cs

```diff
@@ -120,6 +120,26 @@ public bool TryUnpack<T>(out T result) where T : IMessage, new()
             return true;
         }
 
+        /// <summary>
+        /// Attempts to unpack the content of this Any message into one of the message types
+        /// in the given type registry, based on the type URL.
+        /// </summary>
+        /// <param name="registry">The type registry to consult for messages.</param>
+        /// <returns>The unpacked message, or <c>null</c> if no matching message was found.</returns>
+        public IMessage UnpackMessage(TypeRegistry registry)

```

Comment:
Why UnpackMessage instead of Unpack?

Reply:
(I'm guessing having multiple Unpack overloads with different return types could be confusing)

Reply:
Precisely your guess. Although that doesn't mean I'm unwilling to discuss it :)

Reply:
I don't know how much president there is for different return types in overloads from .NET APIs or ASP.NET Core APIs. Is there is an example of a popular API that has different return types? I can't think of one off the top of my head.

Reply:
There's some precedent for this, in terms of `Enum.Parse` vs `Enum.Parse<TEnum>`. But it's relatively rare.

Reply:
IMO I'd name them the same. They both unpack a message, just the arguments to do it (generic vs TypeRegistry) are different.

Reply:
+1 to Unpack.  Both of these methods "unpack a message" as James said, so I find it more confusing for only one of them to be named "UnpackMessage".  I supposed something like "UnpackFromTypeRegistry" would work too, but that seems a bit verbose

Reply:
Renamed to Unpack.